File: My_Notes/Programming/Samples/async_samples/ts/sync.ts
import { DBEngineer, APIEngineer, UIEngineer } from "./engineer";

export function sync(): void {
    var job = new Job(3,1,2);
    var dbEngineer = new DBEngineer();
    var apiEngineer = new APIEngineer();
    var uiEngineer = new UIEngineer();
    dbEngineer.do(job).then(
        ()=>{apiEngineer.do(job).then(
            ()=>{uiEngineer.do(job)}
        );}
    );
}

File: My_Notes/Programming/Samples/async_samples/dart/sync.dart
import './job.dart';
import './engineer.dart';

void sync() {
  Job job = Job(3, 1, 2);
  DBEngineer dbEngineer = DBEngineer();
  APIEngineer apiEngineer = APIEngineer();
  UIEngineer uiEngineer = UIEngineer();
  dbEngineer.doTask(job);
  apiEngineer.doTask(job);
  uiEngineer.doTask(job);
}
